CMakeLists.txt

```diff
@@ -0,0 +1,48 @@
+#=============================================================================
+# Copyright 2022 NVIDIA Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#=============================================================================
+
+cmake_minimum_required(VERSION 3.22.1 FATAL_ERROR)
+
+##############################################################################
+# - Download and initialize RAPIDS CMake helpers -----------------------------
+
+if(NOT EXISTS ${CMAKE_BINARY_DIR}/RAPIDS.cmake)
+  file(DOWNLOAD https://raw.githubusercontent.com/rapidsai/rapids-cmake/branch-22.08/RAPIDS.cmake
+       ${CMAKE_BINARY_DIR}/RAPIDS.cmake)
+endif()
+include(${CMAKE_BINARY_DIR}/RAPIDS.cmake)
+include(rapids-cmake)
+include(rapids-cpm)
+include(rapids-cuda)
+include(rapids-export)
+include(rapids-find)
+
+set(legate_core_version 22.07.00)
+
+if(NOT SKBUILD)
+  project(legate_core VERSION ${legate_core_version} LANGUAGES C CXX)
+  include(legate_core_cpp.cmake)
+else()
+  project(
+    legate_core_python
+    VERSION ${legate_core_version}
+    LANGUAGES # TODO: Building Python extension modules via the python_extension_module requires the C
+              # language to be enabled here. The test project that is built in scikit-build to verify
+              # various linking options for the python library is hardcoded to build with C, so until
+              # that is fixed we need to keep C.
+              C CXX)
+  include(legate_core_python.cmake)
```

Comment:
Overall, I like the Python/C++ split here. The flow got a bit awkward to me since we start in CMakeLists.txt and then re-visit the same file with the `add_subdirectory(...)` call. I think the `legate_core_cpp.cmake` file makes sense to have, but maybe we can refactor this to eliminate the `legate_core_python.cmake`.

roughly like this so it's a linear path through the CMake files
```
if (SKBUILD)
 Python stuff....
 if (FIND_LEGATE_CORE_CPP)
 else()
   include(legate_core_cpp)
 endif()
else()
  include(legate_core_cpp)
endif()

```

Reply:
I expect `legate_core_python.cmake` to grow in size, especially if we start adding Cython things, so I'd prefer to keep them separate.

As for why I'm doing `add_subdirectory` vs. `include()` -- scikit build makes its build dir `_skbuild/<arch-specific-string>/cmake-build`. Calling `add_subdirectory(. build)` with `SKBUILD` temporarily set to false means the C++ library will always land in `build/lib` (along with the legate_core C++ package metadata).

This has the nice side-effect of ensuring regardless of whether you configured the C++ yourself, or let scikit-build do it for you, the C++ build dir is always `build`.

Reply:
Sounds good. Not a major complaint. Better to merge the current PR and get something working and worry about nit clean up later.